addons/website/static/src/snippets/s_instagram_page/options.js

```diff
@@ -0,0 +1,119 @@
+        if (!widgetValue) {
+            this.displayNotification({
+                message: _t("The Instagram page name is not valid"),
+                type: "warning",
+            });
+        }
+        this.$target[0].dataset.instagramPage = widgetValue || "";
+        // As the public widget restart is disabled for instagram, we have to
+        // manually restart the widget.
+        this.trigger_up("widgets_start_request", {
+            $target: this.$target,
+            editableMode: true,
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeWidgetState(widgetName, params) {
+        if (widgetName === "setInstagramPage") {
+            return this.$target[0].dataset.instagramPage;
+        }
+        return this._super(...arguments);
+    },
+    /**
+     * Returns the instagram page name from the DB (or the cache).
+     *
+     * @private
+     * @returns {string|undefined}
+     */
+    async _getDbInstagramPageName() {
+        const igPageInDb = await this._getDbSocialInstagram();
+        if (!igPageInDb) {
+            return;
+        }
+        return this._getInstagramPageNameFromUrl(igPageInDb);
+    },
+    /**
+     * Returns the value of social_instagram from the DB (or the cache).
+     *
+     * @private
+     * @returns {string|undefined}
+     */
+    async _getDbSocialInstagram() {
+        // First we check if the user has changed his instagram during the
+        // current edition (via the social media options).
+        const dbSocialValuesCache = SocialMediaOption.getDbSocialValuesCache();
+        if (dbSocialValuesCache && dbSocialValuesCache["social_instagram"]) {
+            return dbSocialValuesCache["social_instagram"];
+        }
+        // If not, we check the value in the DB.
+        let websiteId;
+        this.trigger_up("context_get", {
+            callback: function (ctx) {
+                websiteId = ctx["website_id"];
+            },
+        });
+        const values = await this._rpc({
+            model: "website",
+            method: "read",
+            args: [websiteId, ["social_instagram"]],
+        });
+        return values[0]["social_instagram"];
+    },
+    /**
+     * Returns the instagram page name from the given url.
+     *
+     * @private
+     * @param {string} url
+     * @returns {string|undefined}
+     */
+    _getInstagramPageNameFromUrl(url) {
+        const preNameString = "instagram.com/";
+        const pageName = url.split(preNameString)[1];
+        if (!pageName || pageName.includes("?") || pageName.includes("#") ||
+            (pageName.includes("/") && pageName.split("/")[1].length > 0)) {
+            return;
+        }
+        return pageName.split("/")[0];
+    },
+});
+
+export default {
+    InstagramPage: options.registry.InstagramPage,
+};
```

Comment:
Should not we rather export the class directly?

```
export default InstagramPage = options.class.blablabla;
options.registry.InstagramPage = InstagramPage;
```
Something like that?

Reply:
We can yes but it would be strange no, it would be the only option exported this way ?